azure-toolkit-libs/azure-toolkit-auth-lib/src/main/java/com/microsoft/azure/tools/auth/core/profile/VisualStudioCodeProfileRetriever.java

```
@@ -0,0 +1,118 @@
+
+            JsonNode userSettings = getUserSettings();
+            if (Objects.isNull(userSettings)) {
+                return null;
+            }
+            Map<String, String> details = getUserSettingsDetails(userSettings);
+            String cloud = details.get(CLOUD);
+            if (StringUtils.isBlank(accessor.getCredentials("VS Code Azure", cloud))) {
+                return null;
+            }
+            return getVsCodeAccountProfile(userSettings, cloud);
+        } catch (CredentialUnavailableException | InvalidConfigurationException exception) {
+            return null;
+        }
+    }
+
+    private static Map<String, String> getUserSettingsDetails(JsonNode userSettings) {
+        Map<String, String> map = new HashMap<>();
+        if (userSettings != null && !userSettings.isNull()) {
+            if (userSettings.has(AZURE_CLOUD) && StringUtils.isNotBlank(userSettings.get(AZURE_CLOUD).asText())) {
+                map.put(CLOUD, userSettings.get(AZURE_CLOUD).asText());
+            }
+            if (userSettings.has(AZURE_TENANT) && StringUtils.isNotBlank(userSettings.get(AZURE_TENANT).asText())) {
+                map.put(TENANT, userSettings.get(AZURE_TENANT).asText());
+            }
+        }
+        map.putIfAbsent(CLOUD, "AzureCloud");
+        return map;
+    }
+
+    private static VisualStudioCodeAccountProfile getVsCodeAccountProfile(JsonNode userSettings, String cloud) throws InvalidConfigurationException {
+        List<String> filteredSubscriptions = new ArrayList<>();
+
+        if (userSettings.has(AZURE_RESOURCE_FILTER)) {
+            for (JsonNode filter : userSettings.get(AZURE_RESOURCE_FILTER)) {
+                String[] tenantAndSubsId = StringUtils.split(filter.asText(), "/");
+                if (tenantAndSubsId.length == 2) {
+                    filteredSubscriptions.add(tenantAndSubsId[1]);
+                } else {
+                    throw new InvalidConfigurationException(
```

this exception is swallowed by the outer `catch`